packages/react/src/global-nav-item/GlobalNavItem.tsx

```diff
@@ -0,0 +1,57 @@
+import * as RadixCollapsible from "@radix-ui/react-collapsible";
+import { forwardRef, useContext } from "react";
+
+import { Box, type BoxProps } from "../box";
+import { Button } from "../button";
+import { Flex } from "../flex";
+import { GlobalNavContext } from "../global-nav-context";
+import { extractSprinkles } from "../sprinkles";
+import * as styles from "./GlobalNavItem.css";
+
+export type GlobalNavItemProps = BoxProps<
+  typeof Button,
+  {
+    active?: boolean;
+  }
+>;
+
+export const GlobalNavItem = forwardRef<HTMLButtonElement, GlobalNavItemProps>(
+  ({ active, children, endDecorator, startDecorator, ...props }, ref) => {
+    const { restProps, sprinkleProps } = extractSprinkles(props);
+
+    const { open } = useContext(GlobalNavContext);
+
+    return (
+      <Flex
+        asChild
+        {...styles.item({
+          active,
+        })}
+        {...sprinkleProps}
+      >
+        <Button
+          appearance="secondary"
+          icon={startDecorator}
+          ref={ref}
+          {...restProps}
+        >
```

Comment:
We can remove the css.ts and have inline styling as button has an `active` state.

```suggestion
      <Flex
        alignItems="center"
        asChild
        flexDirection="row"
        gap="sm"
        h="lg"
        px="sm"
        py="xs"
        w="full"
        {...sprinkleProps}
      >
        <Button
          appearance="secondary"
          data-state={active ? "active" : undefined}
          icon={startDecorator}
          ref={ref}
          {...restProps}
        >
```

Reply:
This was interesting. Thanks!